lib/chef/cookbook/metadata.rb

```diff
@@ -618,12 +618,11 @@ def self.from_json(string)
 
       def self.validate_json(json_str)
         o = Chef::JSONCompat.from_json(json_str)
-        metadata = new()
         VERSION_CONSTRAINTS.each do |dependency_type, hash_key|
           if dependency_group = o[hash_key]
            dependency_group.each do |cb_name, constraints|
-             if metadata.respond_to?(method_name)
-               metadata.public_send(method_name, cb_name, *Array(constraints))
+             if self.respond_to?(dependency_type)
+               self.public_send(dependency_type, cb_name, *Array(constraints))
```

Comment:
I'm pretty certain you don't want to change from doing this against an instance to doing this against the class (its a class method so 'self' is the class).

I think what is going on is that the respond_to? is normally a worthless check that always passes, since you're looping over a constant that the class has set on it, and all the dependency_types that you loop over are valid methods on the instances.  But when you try to use 'self' then the respond_to check silently masks all the NoMethodErrors that you would otherwise be getting.  So I think if you drop the respond_to? check entirely that you'll see those failures.